conn.go

```diff
@@ -202,21 +202,36 @@ func ParseConfig(connString string) (*ConnConfig, error) {
 		}
 	}
 
-	var topologyKeys []string = nil
+	var topologyKeys map[int][]string = nil
 	if s, ok := config.RuntimeParams["topology_keys"]; ok {
 		delete(config.RuntimeParams, "topology_keys")
 		if tkeys, err := validateTopologyKeys(s); err == nil {
-			topologyKeys = tkeys
-		} else {
-			return nil, fmt.Errorf("invalid topology_keys: %v", err)
+			topologyKeys = make(map[int][]string)
+			for _, tk := range tkeys {
+				zones := strings.Split(tk, ":")
+				if len(zones) == 1 {
+					topologyKeys[0] = append(topologyKeys[0], zones[0])
+				} else {
+					num, err := strconv.Atoi(zones[1])
+					if err != nil {
+						return nil, fmt.Errorf("invalid PREFERENCE_VALUE")
```

Comment:
Please use a consistent error message as far as possible, so the one below could be used here as well.